### DISCUSSION TITLE
MRTK v3.0.0 GA

### DISCUSSION THREAD

**Discussion by AMollis** [MixedRealityToolkit/MixedRealityToolkit-Unity#452] [Releases]
![MRTK3_Release_Image_v3_0_0](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/36461279/fa73e9fe-88f5-4cbb-9c0e-aa3a1508171d)

The Mixed Reality Toolkit Organization is happy to announce the general availability of MRTK3. This has been a huge effort by many, and it is inspiring to see this milestone come to completion. Thank you to the founding MRTK members and the new team for all your amazing work. Also, thank you to the community for your help, we couldn't have reached this moment without you.

We recommend that applications use Microsoft's Mixed Reality Feature Tool to obtain the latest version of MRTK3. Follow Microsoft's instructions for [setting up a new MRTK3 project](https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk3-overview/getting-started/setting-up/setup-new-project), these steps include instructions for using the Mixed Reality Feature Tool.

Please note, this will be the final "multi-package" release for MRTK.  Future releases will be done on a per package basis. 

As always, we welcome your feedback and look forward to working together to make MRTK even better!

# Changelog for MRTK v3.0.0

## [v3.0.0] - 2023-9-6 - [`48686f9`](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/commit/48686f91dbb64fc22514e2a0083aab47565be243)

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/). The content is mirrored at [the changelog page of the Microsoft docs website](https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk3-overview/changelogs/mrtk3-v3-0-0-changelog), for now.
  
### Fixed

- Fixed scripts for generating release packages. ([37](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/37), [427](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/427), [434](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/434), [445](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/445))
- Fixed `ExperimentalDrawer` so it compiles in Unity Editor 2022.3.2 and above. ([40](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/issues/40))
- Fixed project template for new project. ([446](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/446))
- MRTK3 now compiles in Unity 2022 without automated code changes. ([11668](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11668))
  

<hr /><em>This discussion was created from the release <a href='https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/releases/tag/'>MRTK v3.0.0 GA</a>.</em>

**Comment by LocalJoost** [comment]
Hi, are you aware the _upgrade_ feature of the MRTK feature tool is not longer working properly with this release? If I select an existing project, now on pre-17, it doesn't register the pre-17 packages being installed (didn't even have time to upgrade to pre-18)
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/4129183/e7bf651d-42b1-4d7c-a666-5cffd6c4d493)
If I install 3.0.0GA the new packages are added, but the old ones not deleted (like what happened before), so I end up with all kinds of duplicate definition errors. You might want to address that, or at least mention that in documentation.

Also, changing all the name spaces from Microsoft.MixedReality.Blah to MixedReality.Blah will make a lot of people unhappy. I understand why, but it's a bit late in the release cycle. I seem to get away with a search & replace.

**Reply by shaynie** [reply]
@LocalJoost , In the feature tool settings, there is a setting to show deprecated packages.  If you select that, you will see new com.microsoft.mrtk.* packages that bring in the new org.mixedrealitytoolkit.* packages.  That is what we decided to provide for an upgrade path.  Please let me know if it works for you.

Thanks,
Scott

**Reply by AMollis** [reply]
@LocalJoost ,  the pre.18 packages will help with the upgrade path.  Hopefully people caught this in the previous release notes, https://github.com/orgs/MixedRealityToolkit/discussions/426

To avoid new user confusion, we decided it best to hide the old packages under the deprecation flag (as mentioned by @shaynie).  Unfortunately, the MRFT does not show deprecated packages that are already installed, if the deprecation flag is not set.

![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/36461279/8f3442d9-5ea9-47ad-aa57-e43801d8444c)

**Reply by LocalJoost** [reply]
Thank you for this clarifications. Since the previous .18 release was so short ago, I think people might have missed it - I did. I am not sure if showing the deprecated packages will help, but anyway, now at least people can see directly under the announcement what they might have missed and what the consequences and necessary actions following of upgrading are. In the mean time, THANKS! We were really waiting for this GA!

**Comment by Suman-298** [comment]
Will Mrtk 3 work as cross platform for Oculus quest 2, windows, mobile and HoloLens 2.

**Reply by LocalJoost** [reply]
@Suman-298 as you might have seen on my blog previous versions worked on all of them, and Magic Leap 2 as well. This stuff is _pretty_ cross platform

**Comment by LocalJoost** [comment]
@AMollis you are you aware of the fact that after upgrading successfully to MRTK3 GA, which I did, the MRKT Feature Tool does no longer recognize the installed packages. So if you want to install additional packages, you will need to remember which one you already installed, or look in the manifest file yourself. I am using FT 1.0.2209.0, downloaded Sept 8. (ie 2 days ago)
